### ISSUE TITLE
Hiding Fields in Serializers

### ISSUE THREAD

**Issue by savitmk** [tfranzel/drf-spectacular#100]
What would be the best method for hiding a subset of fields on a serializer?

**Comment by tfranzel** [comment]
there is no mechanism for that. It is likely doable with a custom Extension (https://drf-spectacular.readthedocs.io/en/latest/customization.html#declare-serializer-magic-with-openapiserializerextension) and if not with a postprocessing hook (step 5). not sure if its worth the effort though.

**Comment by savitmk** [comment]
I tried creating a custom serializer field (e.g. `MyField`) and annotating it with `@extend_schema_field(OpenApiTypes.NONE)`, which does not cause `MyField` to appear in the Swagger example request body, but `MyField` _does_ appear in the request schema with an empty definition, `MyField: {}`.

Would it be incorrect for fields marked `OpenApiTypes.NONE` to not appear at all? Could `OpenApiTypes` include an `IGNORE` option which would mean the field is not written to the schema?

**Comment by tfranzel** [comment]
> but MyField does appear in the request schema with an empty definition, MyField: {}.

i tried reproducing your statement with a test but was unable to https://github.com/tfranzel/drf-spectacular/commit/fcee218876c4ae4a700cceb09fe3dfaf01238817#diff-05942919b2ed60a337f71617b2dbce8fR698

> Would it be incorrect for fields marked OpenApiTypes.NONE to not appear at all
i agree and that was the original intention. if you can still produce this, please provide a code snippet for reproduction.
 
i also did a small improvement to ``OpenApiTypes.NONE``. it should now be interchangable with ``None`` everywhere for convenience.

> What would be the best method for hiding a subset of fields on a serializer?

i gave this a bit more thought and i think it is a reasonable feature. in fact it was a tiny change. ``extend_schema_serializer(exclude_fields=['MyField'])`` should do exactly what you want now.

**Comment by tfranzel** [comment]
@savitmk could you test the changes and comment on the issue reproduction? thanks.

**Comment by savitmk** [comment]
> extend_schema_serializer(exclude_fields=['MyField'])

this is perfect, thank you!

> comment on the issue reproduction

```python3
@extend_schema_field(OpenApiTypes.NONE)
class MyField(serializers.Field):
    pass
```

In my serializer:

```python3
class ExampleSerializer(serializers.Serializer):
    my_field = MyField()
```

On latest release `v0.9.8`, in the result swagger UI, in "Example Value" I do not see `my_field`. When I click "Schema" I see `my_field: {}`.

On latest `master`, I do not see `my_field` in the schema.

**Comment by tfranzel** [comment]
i tracked the bug. it was already fixed in `0.9.9` with 24fcef9. it was targeted at another problem but it also fixed this one. i think we can call this resolved.
